plugin/sampling/strategystore/static/strategy_store_test.go

```diff
@@ -346,38 +365,48 @@ func TestAutoUpdateStrategy(t *testing.T) {
 		time.Sleep(1 * time.Millisecond)
 	}
 	assert.EqualValues(t, makeResponse(sampling.SamplingStrategyType_PROBABILISTIC, 0.9), *s)
+}
 
-	// Test auto update strategy with URL option.
+func TestAutoUpdateStrategyWithURL(t *testing.T) {
 	mockServer := mockStrategyServer()
-	ss, err = NewStrategyStore(Options{
+	ss, err := NewStrategyStore(Options{
 		StrategiesFile: mockServer.URL,
 		ReloadInterval: 10 * time.Millisecond,
 	}, zap.NewNop())
 	require.NoError(t, err)
-	store = ss.(*strategyStore)
+	store := ss.(*strategyStore)
 	defer store.Close()
 
-	// copy existing fixture content to restore it later.
-	srcBytes, err = ioutil.ReadFile(srcFile)
-	require.NoError(t, err)
-	originalBytes := srcBytes
-
 	// confirm baseline value
-	s, err = store.GetSamplingStrategy(context.Background(), "foo")
+	s, err := store.GetSamplingStrategy(context.Background(), "foo")
 	require.NoError(t, err)
 	assert.EqualValues(t, makeResponse(sampling.SamplingStrategyType_PROBABILISTIC, 0.8), *s)
 
 	// verify that reloading in no-op
-	value = store.reloadSamplingStrategy(samplingStrategyLoader(mockServer.URL), string(srcBytes))
-	assert.Equal(t, string(srcBytes), value)
-
-	// update original strategies file with new probability of 0.9
-	newStr = strings.Replace(string(srcBytes), "0.8", "0.9", 1)
-	require.NoError(t, ioutil.WriteFile(srcFile, []byte(newStr), 0644))
-	defer func() {
-		// replace original strategies file with old content.
-		require.NoError(t, ioutil.WriteFile(srcFile, originalBytes, 0644), "failed to restore original file content")
-	}()
+	value := store.reloadSamplingStrategy(samplingStrategyLoader(mockServer.URL), string(strategiesJSON))
+	assert.Equal(t, string(strategiesJSON), value)
+
+	// update original strategies with new probability of 0.9
+	strategiesJSON = []byte(`
```

Comment:
This is not a good approach to change package variable in-place. Even though it's a var, the general assumption is that package-level vars are constants (in fact I would change its type to string to make a real constant).

An even better approach is to convert it into a function that takes the probability value and returns the full JSON string via Sprintf. Then I would also change makeMockServer to return `(strategy *atomic.String, server *httptest.Server)` (using uber-go/atomic), so that you can manipulate the strategy externally, e.g.:
```go
mockStrategy, server := mockStrategyServer()
...
mockStrategy.Store(strategiesJSON(0.9))
...
```